src/api/License/index.ts

```diff
@@ -6,25 +6,45 @@
 import {Request} from '../../http/request';
 import {LkErrorKey} from '../../http/response';
 
-import {SaveLicenseIfMissingResponse, ISaveLicenseIfMissingParams} from './types';
+import {SaveLicenseResponse, ISaveLicenseParams, LicenseInfo} from './types';
 
 export * from './types';
 
-const {INVALID_LICENSE} = LkErrorKey;
+const {INVALID_LICENSE, FORBIDDEN, UNAUTHORIZED} = LkErrorKey;
 
 export class LicenseAPI extends Request {
   /**
-   * Verify the submitted license and save it if it’s valid and there is no saved license yet.
+   * Get information about the currently saved license.
    */
-  public saveLicenseIfMissing(
-    this: Request<SaveLicenseIfMissingResponse>,
-    params: ISaveLicenseIfMissingParams
-  ) {
+  public getLicenseInfo(this: Request<LicenseInfo>) {
     return this.request({
-      errors: [INVALID_LICENSE],
+      errors: [FORBIDDEN, UNAUTHORIZED],
+      url: '/license',
+      method: 'GET'
+    });
+  }
+
+  /**
+   * Verify the submitted license and save it if it's valid and there is no saved license yet.
+   */
+  public saveLicenseIfMissing(this: Request<SaveLicenseResponse>, params: ISaveLicenseParams) {
+    return this.request({
+      errors: [INVALID_LICENSE, UNAUTHORIZED],
       url: '/license',
       method: 'POST',
       params: params
     });
   }
+
+  /**
+   * Replace the saved license with the submitted one. Fails if submitted license is invalid.
+   */
+  public updateLicense(this: Request<SaveLicenseResponse>, params: ISaveLicenseParams) {
```

Comment:
I think it is better to return LicenseInfo here

Reply:
not blocking (not sure if it is possible)

Reply:
Discussed all together: OK to implement Amine's proposal, this avoids the frontend sending an additional request to retrieve the updated `endDate` value.

Reply:
Done :ok_hand: